### ISSUE TITLE
Parameters are not displayed in the structure view when I set a default assignment for a parameter

### ISSUE THREAD

**Issue by leobm** [Camelcade/Perl5-IDEA#2411]
Hi, when I read in the function parameters via shift and I set a default value via the || notation, the parameters are no longer displayed in the structure view. Is it possible to fix this?

```
sub exampleGetProductImageByID {

    my $param_sCompanyID = shift;
    my $param_sProductID = shift;
....
```
![grafik](https://user-images.githubusercontent.com/25712/124962223-ab268080-e01e-11eb-89ad-8fbca7b2bbee.png)

```
sub exampleGetProductImageByID {

    my $param_sCompanyID = shift || "oode";
    my $param_sProductID = shift || "335418GN";
....
```
![grafik](https://user-images.githubusercontent.com/25712/124962345-ca251280-e01e-11eb-94c6-d0e6ce04b466.png)

regards, 

Jan-Felix

**Comment by hurricup** [comment]
Yep, stumbled on this. Feels like not very easy, but yes, doable

**Comment by targzeta** [comment]
It happens also with the defined-or operator (‘//’)

Best regards,
Emanuele

**Comment by hurricup** [comment]
yep

**Comment by leobm** [comment]
> defined-or operator

oh, I didn't know the operator at all. It is often more useful than the || operator.

**Comment by leobm** [comment]
Hi,

I would have another idea.
How about if there would be an **annotation** (maybe @parameter) with which I could define a variable as a parameter?
because in practice it will probably be really difficult to cover all cases. See for example my example.

```perl
#@parameter
my $param_refHashOrArray = shift;

#@parameter
my $param_arefKeyPath = IsArray($_[0]) ? $_[0] : [ @_ ]; ## optional
```

**Comment by hurricup** [comment]
@leobm how do you see this case in a form of signature?
```
somemethod($param_refHashOrArray, ???);
```
